experimental/timezone/src/time_zone.rs

```diff
@@ -2,39 +2,31 @@
 // called LICENSE at the top level of the ICU4X source tree
 // (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
 
-use crate::provider::time_zones::{MetaZoneId, TimeZoneBcp47Id};
+use crate::provider::{MetaZoneId, TimeZoneBcp47Id};
 use tinystr::TinyStr8;
 
-use crate::date::*;
+use crate::{GmtOffset, TimeZoneError};
 use core::str::FromStr;
 
-/// A temporary struct that implements [`TimeZoneInput`]
-/// and is used in tests, benchmarks and examples of this component.
-///
-/// *Notice:* Rust at the moment does not have a canonical way to represent time zones. We are introducing
-/// [`MockTimeZone`](crate::mock::time_zone::MockTimeZone) as an example of the data necessary for
-/// ICU [`TimeZoneFormatter`](crate::TimeZoneFormatter) to work, and [we hope to work with the community](
-/// https://github.com/unicode-org/icu4x/blob/main/docs/research/datetime.md) to develop core date and time
-/// APIs that will work as an input for this component.
+/// A utility type that can hold time zone information
 ///
 /// # Examples
 ///
 /// ```
-/// use icu::datetime::date::GmtOffset;
-/// use icu::datetime::mock::time_zone::MockTimeZone;
+/// use icu::timezone::{GmtOffset, TimeZone};
 ///
-/// let tz1 = MockTimeZone::new(
+/// let tz1 = TimeZone::new(
 ///     GmtOffset::default(),
 ///     /* time_zone_id */ None,
 ///     /* metazone_id */ None,
 ///     /* time_variaint */ None,
 /// );
 ///
-/// let tz2: MockTimeZone = "+05:00".parse().expect("Failed to parse a time zone.");
+/// let tz2: TimeZone = "+05:00".parse().expect("Failed to parse a time zone.");
 /// ```
 #[derive(Debug, Default)]
 #[allow(clippy::exhaustive_structs)] // this type will not add fields (it is largely an example type)
-pub struct MockTimeZone {
+pub struct TimeZone {
```

Comment:
Nit: We discussed offline that we should name this `CustomTimeZone` or `RawTimeZone` or something, in order to reserve the `TimeZone` name for something that has nicer ways to perform its own calculations.

I quite like `CustomTimeZone` because that's really what this type is. You can put arbitrary custom values in to the fields, and ICU4X will make a best attempt at formatting them.

Reply:
I'm not convinced; my hope for this crate was that this is the canonical icu4x type and instead we have a TimeZoneResolver object that manipulates them

but I'll make the change, it's easier to revert later.